decidim-proposals/app/controllers/decidim/proposals/proposal_votes_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # Exposes the proposal vote resource so users can vote proposals.
+    class ProposalVotesController < Decidim::Proposals::ApplicationController
+      before_action :authenticate_user!
+
+      def create
+        @proposal = Proposal.find(params[:proposal_id])
+
+        set_votes_count_classes
```

Comment:
I'm agree. The problem is I'm using the `request.referrer` for that so I think I can't use a helper, right?